src/tribler/gui/core_manager.py

```diff
@@ -47,8 +51,8 @@ def __init__(self, root_state_dir: Path, api_port: int, api_key: str, app_manage
         self.should_quit_app_on_core_finished = False
 
         self.use_existing_core = True
-        self.last_core_stdout_output: str = ''
-        self.last_core_stderr_output: str = ''
+        self.last_core_stdout_output: deque = deque(maxlen=CORE_OUTPUT_DEQUE_LENGTH)
+        self.last_core_stderr_output: deque = deque(maxlen=CORE_OUTPUT_DEQUE_LENGTH)
```

Comment:
The last write to stderr does not necessarily contain the actual the actual exception traceback, so we store several writes.